libfwupd/meson.build

```diff
@@ -99,6 +106,33 @@ if get_option('enable-introspection')
     ],
     install : true
   )
+
+  # Verify the map file is correct -- note we can't actually use the generated
+  # file for two reasons:
+  #
+  #  1. We don't hard depend on GObject Introspection
+  #  2. The map file is required to build the lib that the GIR is built from
+  #
```

Comment:
I think using the generated file and making it automagic would really be nice.
Maybe a crazy idea, but let me run it by you.
1. Make GObject introspection a hard dependency
2. Build the library that the GIR is built from twice.  First build it without a map file (or an empty/dummy one), build the GIR, and then rebuild it with the map file.

Reply:
I've talked to a few people shipping fwupd about GObject introspection being mandatory; it didn't go down well. I think GObject introspection and cross compiling are not always friends. Also, from a project point of view I think it makes sense to have a documented API rather than generating it from `Since` tags; it's a cute hack, but I'm guessing most people won't realize changing a "comment" would change the library ABI.

Reply:
OK.